src/views/RecIM/components/Forms/SeriesScheduleForm/index.jsx

```diff
@@ -0,0 +1,331 @@
+  ].flat();
+
+  const currentInfo = useMemo(() => {
+    return {
+      SeriesID: seriesID,
+      Sun: false,
+      Mon: false,
+      Tue: false,
+      Wed: false,
+      Thu: false,
+      Fri: false,
+      Sat: false,
+      AvailableSurfaceIDs: [],
+      DailyStartTime: '',
+      DailyEndTime: '',
+      EstMatchTime: '',
+    };
+  }, [seriesID]);
+
+  const [newInfo, setNewInfo] = useState(currentInfo);
+  const [openConfirmWindow, setOpenConfirmWindow] = useState(false);
+  const [isSaving, setSaving] = useState(false);
+  const [disableSubmitButton, setDisableSubmitButton] = useState(true);
+
+  const handleSetError = (field, condition) => {
+    const getCurrentErrorStatus = (currentValue) => {
+      return {
+        ...currentValue,
+        [field]: condition,
+      };
+    };
+    setErrorStatus(getCurrentErrorStatus);
+  };
+
+  const isNumeric = (value) => {
+    return /^-?\d+$/.test(value) || value.length === 0;
+  };
+  // Field Validation
+  useEffect(() => {
+    let hasError = false;
+    let hasChanges = false;
+    for (const field in currentInfo) {
+      if (currentInfo[field] !== newInfo[field]) {
+        hasChanges = true;
+      }
+      //switch case used here TEMPORARILY as a template for other forms
+      switch (field) {
+        case 'EstMatchTime':
+          handleSetError(field, !isNumeric(newInfo[field]));
+          hasError = !isNumeric(newInfo[field]) || hasError;
+          break;
+        default:
+          handleSetError(field, newInfo[field] === '');
+          hasError = newInfo[field] === '' || hasError;
+          break;
+      }
+    }
+    setDisableSubmitButton(hasError || !hasChanges);
+  }, [newInfo, currentInfo]);
+
+  const handleChange = (event, src) => {
+    const getNewInfo = (currentValue) => {
+      // datetime pickers return value rather than event,
+      // so we can also manually specify target source and value
+      if (src) {
+        let newValue = event;
+        return {
+          ...currentValue,
+          [src]: newValue,
+        };
+      }
+      return {
+        ...currentValue,
+        [event.target.name]:
+          event.target.type === 'checkbox' ? event.target.checked : event.target.value,
+      };
+    };
+    setNewInfo(getNewInfo);
+  };
+
+  const getFieldLabel = (fieldName) => {
+    const matchingField = allFields.find((field) => field.name === fieldName);
+    return matchingField.label;
+  };
+
+  function getNewFields(currentInfo, newInfo) {
+    const updatedFields = [];
+    Object.entries(newInfo).forEach(([key, value]) => {
+      if (currentInfo[key] !== value)
+        updatedFields.push({
+          Field: key,
+          Value: value,
+          Label: getFieldLabel(key),
+        });
+    });
+    return updatedFields;
+  }
+
+  const handleConfirm = () => {
+    setSaving(true);
+
+    let dayOfTheWeek = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
```

Comment:
It would technically be more efficient to put this as a const outside of the component itself. Otherwise, it is getting re-created every time `handleConfirm` is called (which may only be once anyway, but it's good practice to do it consistently)